### PR TITLE
Add remote driver guide

### PR DESCRIPTION
**Pull request by jedevc**
Added basic scenarios for:
- Connecting over a basic unix socket
- Connecting to a docker buildkit container
- Connecting to a kubernetes buildkit deployment

### CODE REVIEW

**Comment by jedevc** [comment]
@crazy-max have updated to follow the guide to follow the new-style docs (and tested to make sure they'll look alright on docs.docker.com, once https://github.com/docker/docker.github.io/pull/14644 is merged).

File: docs/guides/remote-driver.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+title: "Remote driver"
```
Comment (crazy-max) [inline docs/guides/remote-driver.md:2]: Maybe `Remote builder` would be better

**Review by crazy-max — APPROVED** [review]
LGTM, ptal @tonistiigi @thaJeztah

File: docs/guides/remote-builder.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+We can now connect to this running container using buildx:
+
+```console
+$ docker buildx create \
+  --name remote-kubernetes \
+  --driver remote \
+  --driver-opt cacert=.certs/ca.pem,cert=.certs/client-cert.pem,key=.certs/client-key.pem,servername=... \
+  tcp://buildkitd.default.svc:1234
+```
```
Comment (AkihiroSuda) [inline docs/guides/remote-builder.md:111]: This seems about Kubernetes, not Docker

File: docs/guides/remote-builder.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+In this scenario, we'll create a setup with buildkitd listening on a unix
+socket, and have buildx connect through it.
+
+Firstly, ensure that [buildkit](https://github.com/moby/buildkit) is installed.
+Then launch an instance of buildkitd:
+
+```console
+$ sudo ./buildkitd --group $(id -gn) --addr unix://$HOME/buildkitd.sock
+```
```
Comment (AkihiroSuda) [inline docs/guides/remote-builder.md:31]: I’d rather recommend using rootless mode, but that can be another PR

Comment (tonistiigi) [reply docs/guides/remote-builder.md:31]: Lets keep it simple as it is and add a link to https://github.com/moby/buildkit/blob/master/docs/rootless.md for rootless. Could even leave the default socket path. Can also point to https://github.com/moby/buildkit/tree/master/examples/systemd for running buildkitd with systemd.

File: docs/guides/remote-builder.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+(remember that `--load` is necessary when not using the default `docker`
+driver, to load the build result into the docker daemon)
+
+## Remote Buildkit in Docker container
+
+In this scenario, we'll create a similar setup to the `docker-container`
+driver, by manually booting a buildkit docker container and connecting to it
+using the buildx remote driver. In most cases you'd probably just use the
+`docker-container` driver directly, but imagine that for some reason you want
```
Comment (tonistiigi) [inline docs/guides/remote-builder.md:76]: > but imagine

Smth like

In most cases, you'd probably just use the `docker-container` driver that connects to buildkit through Docker API, but in this case we are manually creating a container and accessing it with via an exposed port.

File: docs/guides/remote-builder.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+to manually manage the container instead of having buildx manage it for you.
+
+First, we need to generate certificates for buildkit - you can use the 
+[create-certs.sh](https://github.com/moby/buildkit/blob/master/examples/kubernetes/create-certs.sh)
```
Comment (tonistiigi) [inline docs/guides/remote-builder.md:80]: `/blob/master` -> lets use a tagged release link in case the file paths should change.

File: docs/guides/remote-builder.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+allow arbitrary access to buildkit without credentials.
+
+With our certificates generated in `.certs/`, we startup the container:
+
+```console
+$ docker run -d --rm \
+  --name=remote-buildkitd \
+  --privileged \
+  -p 1234:1234 \
+  -v $PWD/.certs:/certs \
```
Comment (tonistiigi) [inline docs/guides/remote-builder.md:92]: Mounting to `/etc/buildkit/certs` might be cleaner

File: docs/guides/remote-builder.md (hunk 3 above)
Comment (jedevc) [reply docs/guides/remote-builder.md:31]: Have setup those links right below this to point users in the right direction - I think we should keep the specified socket path, otherwise we'll also need to tell users later on to use sudo buildx (since the `/var/run/buildkit/` directory is owned by `root:root` and inaccessible, even with `--group`, though maybe that's a bug?).

File: docs/guides/remote-builder.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+socket, and have buildx connect through it.
+
+Firstly, ensure that [buildkit](https://github.com/moby/buildkit) is installed.
+Then launch an instance of buildkitd:
+
+```console
+$ sudo ./buildkitd --group $(id -gn) --addr unix://$HOME/buildkitd.sock
+```
+
+In this example, we run buildkitd as the root user, however, in the real world,
```
Comment (tonistiigi) [inline docs/guides/remote-builder.md:33]: "For example, launch an instance of buildkitd with:

..

Alternatively, [look here] for running buildkitd in rootless mode or [here] for examples of running it as a systems service.

**Review by tonistiigi — APPROVED** [review]
